packages/core/src/internal/decorators/i18n.ts

```diff
@@ -78,13 +81,24 @@ export function i18n() {
 
     const descriptor = {
       get(this: I18nElement) {
-        return { ...(I18nService.keys as any)[this.__i18nKey], ...this.__i18n };
+        const i18nObj = mergeObjects((I18nService.keys as any)[this.__i18nKey], this.__i18n || {});
```

Comment:
The spread operators only do a shallow merge. We have i18n objects now that go more than one level deep. `mergeObjects` does that.